qemu/tests/sr_iov_sanity.py

```diff
@@ -104,35 +111,38 @@ def run(test, params, env):
     ethname_dict = []
     ips = {}
 
-    msg = "Configure all VFs in host."
-    error.context(msg, logging.info)
-    for pci_id in vf_pci_id:
-        cmd = "ls /sys/bus/pci/devices/%s/net/" % pci_id
-        ethname = utils.system_output(cmd).strip()
-        ethname_dict.append(ethname)
-        network_script = os.path.join("/etc/sysconfig/network-scripts",
-                                      "ifcfg-%s" % ethname)
-        if not os.path.exists(network_script):
-            error.context("Create %s file." % network_script, logging.info)
-            txt = "DEVICE=%s\nONBOOT=yes\nBOOTPROTO=dhcp\n" % ethname
-            file(network_script, "w").write(txt)
-
-    msg = "Check whether VFs could get ip in host."
-    error.context(msg, logging.info)
-    for ethname in ethname_dict:
-        ifup_down_interface(ethname)
-        _ip = check_network_interface_ip(ethname)
-        if not _ip:
-            msg = "Interface '%s' could not get IP." % ethname
-            logging.error(msg)
-        else:
-            ips[ethname] = _ip
-            logging.info("Interface '%s' get IP '%s'", ethname, _ip)
+    # Not all test environments would have a dhcp server to serve IP for
+    # all mac addresses. So configure_on_host param has been
+    # introduced to choose whether configure VFs on host or not
+    if configure_on_host:
+        msg = "Configure all VFs in host."
+        error_context.context(msg, logging.info)
+        for pci_id in vf_pci_id:
+            ethname = utils_misc.get_interface_from_pci_id(pci_id)
+            mac = utils_net.generate_mac_address_simple()
+            ethname_dict.append(ethname)
+            try:
+                utils_net.create_network_script(ethname, mac, "dhcp",
+                                                "255.255.255.0", on_boot="yes")
```

Comment:
this will create network script in host, we should consider to clean up the network scripts once the test is done

Reply:
# TODO:  cleanup of the network scripts

Reply:
@balamuruhans cleanup of network scripts to be handled later.